src/commands/runFromPackageDeploy.ts

```diff
@@ -0,0 +1,198 @@
+        const result: MessageItem | undefined = await ext.ui.showWarningMessage(runFromZipDeployMessage, selectStorageAccountButton, DialogResponses.cancel);
+        if (result === selectStorageAccountButton) {
+            const sa: azUtil.IResourceResult = await azUtil.promptForStorageAccount(actionContext, {
+                kind: [],
+                learnMoreLink: 'https://aka.ms/T5o0nf'
+            });
+            if (sa.id) {
+                await updateWorkspaceSetting(settingKey, sa.id, fsPath);
+            }
+            const accountKey: string = 'AccountKey=';
+            name = sa.name;
+            key = sa.connectionString.substring(sa.connectionString.indexOf(accountKey) + accountKey.length);
+        }
+    } else {
+        const sClient: StorageClient = new StorageClient(node.credentials, node.subscriptionId);
+        const rg: string = azUtil.getResourceGroupFromId(storageAccountId);
+        name = azUtil.getNameFromId(storageAccountId);
+        const result: StorageAccountListKeysResult = await sClient.storageAccounts.listKeys(rg, name);
+        if (!result.keys || result.keys.length === 0) {
+            throw new ArgumentError(result);
+        } else {
+            key = result.keys[0].value;
+        }
+    }
+
+    if (name !== undefined && key !== undefined) {
+        return azureStorage.createBlobService(name, key);
+    } else {
+        throw new ArgumentError({ name: name, key: key });
+    }
+}
+
+async function createBlobFromZip(blobService: azureStorage.BlobService, zipFilePath: string, blobName: string): Promise<string> {
+    const containerName: string = 'azureappservice-run-from-zip';
+    await new Promise<void>((resolve: () => void, reject: (err: Error) => void): void => {
+        blobService.createContainerIfNotExists(containerName, (err: Error) => {
+            if (err !== null) {
+                reject(err);
+            } else {
+                resolve();
+            }
+        });
+    });
+
+    await new Promise<void>((resolve: () => void, reject: (err: Error) => void): void => {
+        blobService.createBlockBlobFromLocalFile(containerName, blobName, zipFilePath, (error: Error, _result: azureStorage.BlobService.BlobResult, _response: azureStorage.ServiceResponse) => {
+            if (error !== null) {
+                reject(error);
+
+            } else {
+                resolve();
+            }
+        });
+    });
+    const sasToken: string = blobService.generateSharedAccessSignature(containerName, blobName, <azureStorage.common.SharedAccessPolicy>{
+        AccessPolicy: {
+            Permissions: azureStorage.BlobUtilities.SharedAccessPermissions.READ + azureStorage.BlobUtilities.SharedAccessPermissions.LIST,
+            Start: azureStorage.date.secondsFromNow(-10),
+            // for clock desync
+            Expiry: azureStorage.date.daysFromNow(365),
```

Comment:
I remember having a discussion about this time. Is this the same thing Kudu does for Windows Run From Package?